Source/WebKit/UIProcess/Gamepad/libwpe/UIGamepadProviderLibWPE.cpp

```diff
@@ -27,28 +27,53 @@
 #include "UIGamepadProvider.h"
 
 #if ENABLE(GAMEPAD) && USE(LIBWPE)
-
 #include <WebCore/GamepadProviderLibWPE.h>
+#include <wpe/wpe.h>
 
 #if PLATFORM(WPE)
 #include "WPEWebViewLegacy.h"
+
+#if ENABLE(WPE_PLATFORM)
+#include "GamepadProviderWPE.h"
+#include "WPEWebViewPlatform.h"
+#include <wpe/wpe-platform.h>
+#endif
 #endif
 
+namespace WebKit {
 using namespace WebCore;
 
-namespace WebKit {
+#if ENABLE(WPE_PLATFORM)
+static inline bool usingWPEPlatformAPI()
+{
+    return !!g_type_class_peek(WPE_TYPE_DISPLAY);
+}
```

Comment:
We have a number of variables and similar functions sprinkled around the code (at least I could easily find instances in `WebPasteboardProxyWPE.cpp`, `DisplayVBlankMonitorDRM.cpp`, `WebProcessPoolGLib.cpp`, and `WebKitProtocolHandler.cpp`). While I do not think that is a blocker, we may want to factor them out into a single, shared implementation in the future.

Reply:
Yes, I thought about that, I'll find a common place to move this.